src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala

```diff
@@ -149,21 +161,61 @@ object macros:
     
         case '[JsonObject] =>
           Json.fromJsonObject(JsonObjectUnit)
+
+        case '[Json] =>
+          JsonUnit
+
+        case '[JsonNumber] =>
+          Json.fromJsonNumber(JsonNumberUnit)
     
         case '[t] if TypeRepr.of[t] <:< TypeRepr.of[Product] =>
-          Json.fromFields(deconstructArgumentProduct[t])      
+          Json.fromFields(deconstructArgumentProduct[t])    
     
-        case '[Boolean] => 
+        case '[Boolean] | '[java.lang.Boolean] =>
           Json.fromBoolean(BooleanUnit)
+
+        case '[Byte] | '[java.lang.Byte] =>
```

Comment:
According to 
```
  /**
   * Create a `Json` value representing a JSON number from an `Int`.
   */
  final def fromInt(value: Int): Json = JNumber(JsonLong(value.toLong))

  /**
   * Create a `Json` value representing a JSON number from a `Long`.
   */
  final def fromLong(value: Long): Json = JNumber(JsonLong(value))
  ```
  it's better to combine `Byte`, `Short`, `Int`, `Long` and related java types to one case.